docs/download.md

```diff
@@ -0,0 +1,21 @@
+---
+id: download
+title: Apache Fury(incubating) Download
+---
+
+The official Apache Fury releases are provided as source artifacts.
+
+For binary install, please see Fury [install](/docs/start/install/) document.
+
+## The latest release 
+
+Apache Fury hasn't make a release since joined Apache Incubator.
+
+## Verify a release
+
+Please verify the release with corresponding hashes(sha512), signatures and [release KEYS](https://downloads.apache.org/incubator/fury/KEYS).
```

Comment:
Can I suggest this should be more detailed - a bit like https://pekko.apache.org/download.html#verifying-downloads

Reply:
Thanks, I added a more detailed version.
![image](https://github.com/apache/incubator-fury-site/assets/12445254/f82fe53e-3d05-44c8-bcda-06c208e5133c)

 I was planing to update it to direct to https://fury.apache.org/community/how_to_verify after https://github.com/apache/incubator-fury-site/pull/109 is merged